backend-tests/backend-tests.hs

```diff
@@ -481,6 +484,67 @@ tests wasm_file = testGroup "Tests" $ upgradeGroups $
+
+  , withoutUpgrade $ iiTest "register remote device" $ \cid -> do
+    user_number <- register cid webauth1ID device1 (powAt cid 0) >>= mustGetUserNumber
+    expiration <- callII cid webauth1ID #enter_device_registration_mode user_number
+    V.IsJust (V.Label :: Label "added_tentatively") response <- callII cid anonymousID #add_tentative_device (user_number, device2)
+    lift $ expiration @?= response .! #device_registration_timeout
+    let verification_code = response .! #verification_code
+    V.IsJust (V.Label :: Label "verified") _ <- callII cid webauth1ID #verify_tentative_device (user_number, verification_code)
+    lookupIs cid user_number [device1, device2]
+
+  , withoutUpgrade $ iiTest "should return tentative device" $ \cid -> do
+    user_number <- register cid webauth1ID device1 >>= mustGetUserNumber
+    expiration <- callII cid webauth1ID #enter_device_registration_mode user_number
+    V.IsJust (V.Label :: Label "added_tentatively") _ <- callII cid anonymousID #add_tentative_device (user_number, device2)
+    anchor_info <- callII cid webauth1ID #get_anchor_info user_number
+    lift $ anchor_info .! #devices @?= V.fromList [device1]
+    lift $ anchor_info .! #tentative_device @?= Just device2
+    lift $ anchor_info .! #device_registration_mode_expiration @?= Just expiration
+
+  , withoutUpgrade $ iiTest "reject tentative device if not in registration mode" $ \cid -> do
+    user_number <- register cid webauth1ID device1 >>= mustGetUserNumber
+    V.IsJust (V.Label :: Label "device_registration_mode_off") _ <- callII cid anonymousID #add_tentative_device (user_number, device2)
+    pure ()
+
+  , withoutUpgrade $ iiTest "reject tentative device if registration mode is disabled again" $ \cid -> do
+    user_number <- register cid webauth1ID device1 >>= mustGetUserNumber
+    _ <- callII cid webauth1ID #enter_device_registration_mode user_number
+    _ <- callII cid webauth1ID #exit_device_registration_mode user_number
+    V.IsJust (V.Label :: Label "device_registration_mode_off") _ <- callII cid anonymousID #add_tentative_device (user_number, device2)
+    pure ()
+ 
+  , withoutUpgrade $ iiTest "reject tentative device if registration mode is expired" $ \cid -> do
+    user_number <- register cid webauth1ID device1 >>= mustGetUserNumber
+    expiration <- callII cid webauth1ID #enter_device_registration_mode user_number
+    V.IsJust (V.Label :: Label "added_tentatively") response <- callII cid anonymousID #add_tentative_device (user_number, device2)
+    lift $ expiration @?= response .! #device_registration_timeout
+    setCanisterTimeTo cid (fromIntegral expiration) -- registration mode is expired at time expiration
+    V.IsJust (V.Label :: Label "wrong_code") result <- callII cid webauth1ID #verify_tentative_device (user_number, response .! #verification_code)
+    lift $ result .! #retries_left @?= 0
+
+  , withoutUpgrade $ iiTest "reject wrong code on verify remote device" $ \cid -> do
```

Comment:
nit: we could verify that:
* The device can be verified
* The device can be verified after N retries, with N smaller than MAX_ATTEMPS